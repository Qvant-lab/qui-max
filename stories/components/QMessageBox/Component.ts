// eslint-disable-next-line import/no-extraneous-dependencies
import type { Story } from '@storybook/vue3';
import { defineComponent, defineAsyncComponent, ref } from 'vue';

import type { QMessageBoxProps } from '@/qComponents/QMessageBox';

const QMessageBoxComponentStory: Story<QMessageBoxProps> = () =>
  defineComponent({
    components: {
      MessageBoxFormTest: defineAsyncComponent(
        () => import('./MessageBoxFormTest.vue')
      )
    },
    setup() {
      const isVisible = ref(false);

      const handleClick = async (): Promise<void> => {
        isVisible.value = true;
      };

      return { isVisible, handleClick };
    },
    template: `
      <q-button @click="handleClick">Click to open</q-button>

      <message-box-form-test v-model:isVisible="isVisible"/>
    `
  });

QMessageBoxComponentStory.storyName = 'Component';

export default QMessageBoxComponentStory;
