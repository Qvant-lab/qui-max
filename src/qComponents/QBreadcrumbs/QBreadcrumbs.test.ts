import { describe, it, expect } from 'vitest';
import { shallowMount } from '@vue/test-utils';
import Component from './index';

describe('QBreadcrumbs', () => {
  it('element should match snapshot', async () => {
    const { element } = await shallowMount(Component);

    expect(element).toMatchSnapshot();
  });
});
