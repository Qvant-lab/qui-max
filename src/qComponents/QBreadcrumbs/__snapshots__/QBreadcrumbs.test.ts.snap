// Vitest Snapshot v1

exports[`QBreadcrumbs > element should match snapshot 1`] = `
<div
  class="q-breadcrumbs"
>
  
  
  <div
    class="q-breadcrumbs__crumb q-breadcrumbs__crumb_last"
  />
</div>
`;
